lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,20 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      paths = Set.new(applicable_files.map { |file| File.dirname(file) + File::SEPARATOR })
+      repo_files = Set.new(Overcommit::GitRepo.list_files(paths.to_a) + applicable_files)
+      conflict_hash = repo_files.classify(&:downcase).
+        select { |_, files| files.size > 1 }
+      conflict_files = applicable_files.
+        select { |file| conflict_hash.include? file.downcase }
```

Comment:
This codebase tries to subscribe to [Jim Weirich](https://en.wikipedia.org/wiki/Jim_Weirich)'s approach to parentheses for method calls, which is to include them [if you care about the return value](https://twitter.com/jimweirich/status/373814960788221952) (which we do in this case).